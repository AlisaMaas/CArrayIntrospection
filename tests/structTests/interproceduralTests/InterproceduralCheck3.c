#include "structHeader.h"
/**
 * This simple test should recognize that functions a-e and 
 * find all contain the null-terminated "string".
 **/
void print(char *string) {}
int find(struct foo *f) {
	for (int i = 0;; i++) {
		if (f->string[i] == '\0') {
			print("About to break from loop");
			break;
		}
	}
	return 1;
}
int e(struct foo *f) {
	return find(f);
}
int d(struct foo *f) {
	return e(f);
}
int c(struct foo *f) {
	return d(f);
}
int b(struct foo *f) {
	return c(f);
}
int a(struct foo *f) {
	return b(f);
}
