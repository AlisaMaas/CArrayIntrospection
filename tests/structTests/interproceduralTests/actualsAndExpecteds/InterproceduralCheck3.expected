Printing analysis 'Promote Memory to Register' for function 'print':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'e':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'd':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'c':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'b':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'a':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Determine whether and how to annotate each struct element with the null-terminated annotation':
struct struct.foo[0] should be annotated NULL_TERMINATED (2).
